streamlit_webapp/streamlit_app.py

```
@@ -379,29 +383,36 @@ def clear_dialog():
+            with st.expander("Input your own file", expanded=False):
+                user_files = st.file_uploader("Choose your files", accept_multiple_files=True)
+                if user_files:
+                    filenames = [file.name for file in user_files]
+                    try:
+                        for file in user_files:
+                            file_path = st.session_state.user_input_dir / file.name
+                            with file_path.open("wb") as f:
+                                f.write(file.getbuffer())
+                        st.success(f"File(s) uploaded successfully: {filenames}")
+                        st.session_state.logger.info(f"File(s) uploaded successfully: {filenames}")
+                    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)